how-to-simulate-io-faults-at-runtime.md

```diff
@@ -0,0 +1,137 @@
+---
+title: How to Simulate I/O Faults at Runtime?
+author: ['Keao Yang']
+date: 2021-01-08
+summary: This post dives deep into how we implement the IOChaos experiment without using a sidecar.
+tags: ['Chaos Mesh', 'Chaos engineering', 'Fault injection']
+categories: ['Engineering']
+image: /images/blog/how-to-simulate-io-faults-at-runtime.jpg
+---
+
+![How to simulate I/O faults at runtime](media/how-to-simulate-io-faults-at-runtime.jpg)
+
+In a production environment, filesystem faults might occur due to various incidents such as disk failures and administrator errors. As a Chaos Engineering platform, Chaos Mesh has supported simulating I/O faults in a filesystem ever since its early versions. By simply adding an IOChaos CustomResourceDefinition (CRD), we can watch how the filesystem fails and returns errors.
```

Comment:
```suggestion
In a production environment, filesystem faults might occur due to various incidents such as disk failures and administrator errors. As a Chaos Engineering platform, [Chaos Mesh](https://pingcap.com/blog/chaos-mesh-your-chaos-engineering-solution-for-system-resiliency-on-kubernetes) has supported simulating I/O faults in a filesystem ever since its early versions. By simply adding an IOChaos CustomResourceDefinition (CRD), we can watch how the filesystem fails and returns errors.
```

Reply:
Adding internal links helps our SEO.